tracer/src/Datadog.Trace/ClrProfiler/AutomaticTracer.cs

```diff
@@ -49,15 +51,15 @@ SpanContext IDistributedTracer.GetSpanContext()
 
             var value = DistributedTrace.Value;
 
-            if (value is SpanContext spanContext)
+            return value switch
             {
-                return spanContext;
-            }
-
-            return SpanContextPropagator.Instance.Extract(value);
+                null => null,
+                SpanContext spanContext => spanContext,
+                _ => SpanContextPropagator.Instance.Extract(value!)
```

Comment:
It's a good idea, but the type checker still considers v to `IReadOnlyDictionary<string, string>?`, so doesn't help remove the `!`. (don't know why it looks like it could infer it's not nullable anymore)

Reply:
>It's a good idea, but the type checker still considers v to IReadOnlyDictionary<string, string>?, so doesn't help remove the !. (don't know why it looks like it could infer it's not nullable anymore)

Because you're missing a `?` I think:
- You need this `IReadOnlyDictionary<string, string?>`
- But you have  `IReadOnlyDictionary<string, string>?`